### ISSUE TITLE
Support for Quarkus

### ISSUE THREAD

**Issue by arjantijms** [eclipse-ee4j/mojarra#5442]
See https://stackoverflow.com/questions/59079402/quarkus-extension-or-not-for-jsf

Maybe we could consider Quarkus support at some point.

**Comment by tandraschko** [comment]
i can just say that it requires a lot of work and also changes on existing JSF impl code

IMO it would be cool if you guys would do it but IMO its not worth. MyFaces is running fine already on quarkus.

**Comment by melloware** [comment]
I agree with @tandraschko it will be a lot of effort and maybe not worth your team's time.

**Comment by arjantijms** [comment]
> i can just say that it requires a lot of work and also changes on existing JSF impl code
> 
> IMO it would be cool if you guys would do it but IMO its not worth. MyFaces is running fine already on quarkus.

Maybe something to keep in mind, thanks for chiming in.

I'm also looking for the potentially making Soteria (Jakarta Security) work on Quarkus, which may then be a lot of work as well? Where is most of the work / time spend in making it compatible?

**Comment by tandraschko** [comment]
In short: Quarkus is not fully CDI compatible

some features are comopletely not supported (like e.g. injection in a non-cdi bean), something like Extensions must be completely rewritten in Quarkus API, which is split in build and runtime.

**Comment by tandraschko** [comment]
You can just check our quarkus extensions: https://github.com/apache/myfaces/tree/main/extensions/quarkus

as i said, in addition to make this extension possible, it was needed to refactor some of the JSF Impl code.

**Comment by melloware** [comment]
@arjantijms you can see the Quarkus JSF podcast here for the state of JSF on Quarkus: https://www.youtube.com/watch?v=DIN0I56-GR4

**Comment by BalusC** [comment]
> it was needed to refactor some of the JSF Impl code

To confirm, no changes in JSF API code was needed in MyFaces side? I.e. the code currently in Faces 5.0 branch is fine wrt Quarkus? https://github.com/jakartaee/faces/tree/5.0

**Comment by tandraschko** [comment]
I am almost sure that i only modified impl for quarkus

**Comment by melloware** [comment]
@arjantijms as far as Soteria goes the Quarkus team is super responsive to helping extension developers figure out issues and getting their extension to work with Quarkus.  They have a Zulip channel and I have asked a bunch of questions when I was doing the PF, OmniFaces, Myfaces stuff and almost always got a great answer or another extension sample code to look at for what I was trying to accomplish.

**Comment by arjantijms** [comment]
Thanks, sounds good. I moved it a little higher up on my TODO list ;)

**Comment by BalusC** [comment]
I had Claude take a look and here's its verdict:

---

Took a fresh inventory of what a Quarkus port would actually touch on the Mojarra side, to put numbers on @tandraschko's assessment.

**Impl surface on 5.0 (no API impact — confirms Thomas):**

- 6 portable extensions registered via `META-INF/services`: `CdiExtension`, `FlowCDIExtension`, `FlowDiscoveryCDIExtension`, `ViewScopeExtension`, `ClientWindowScopeExtension`, plus the `ELAwareBeanManagerWrapper` machinery.
- 3 custom CDI contexts: `FlowCDIContext`, `ViewScopeContext`, `ClientWindowScopeContext` → each becomes a Quarkus `ContextRegistrar` contribution.
- `CdiExtension.afterBeanDiscovery()` does ~25 `addBean(new …Producer(beanManager))` calls (FacesContext, ExternalContext, request/session/view maps, flows, cookies, resource handler, …) — all become `SyntheticBeanBuildItem`s.
- On top of that, `ManagedPropertyProducer` is registered **per `@ManagedProperty` type discovered at runtime** via `processAnnotatedType(@WithAnnotations)` — needs to be driven by the Jandex index from a build step instead.

So the cost matches Thomas's description: portable-extension logic splits between a Quarkus deployment module (build steps) and a runtime module, and a chunk of `org.glassfish.mojarra.cdi` / `flow` / `application.view` has to be factored so the synthetic-bean and context wiring is reusable from build steps rather than only callable from `AfterBeanDiscovery`.

**On the "is it worth it" question:** I considered whether a perf-bench angle (same-runtime Mojarra vs MyFaces A/B) tips the balance, but Tomcat + bring-your-own Weld already gives that for free with zero impl work — same servlet container, same Weld, swap only the Faces jar. So the bench justification doesn't really hold up; the cleanest A/B harness is already buildable today with Tomcat.

Which leaves the actual user-facing question: does anyone need Mojarra-on-Quarkus when MyFaces already runs there? Honest answer is "not urgently." Worth keeping open as a standing invitation for a contributor who wants to drive it (a spike on whether stock Mojarra even starts on Quarkus dev-mode would be a cheap first data point), but probably not Mojarra-team-priority work given the impl cost and the perf/spec work that's higher leverage right now.

API stability being confirmed is the good news either way — this stays an impl-side initiative on any timeline, no Faces spec impact.
